django_mongodb_engine/creation.py

```diff
@@ -139,7 +141,17 @@ def create_compound_indexes(indexes, **kwargs):
         # Django unique_together indexes.
         for indexes in getattr(meta, 'unique_together', []):
             assert isinstance(indexes, (list, tuple))
-            create_compound_indexes(indexes, unique=True)
+            """
+            create_compound_indexes by default generates the index name as a list of all the fields in 'indexes',
+            this can cause the index creation to fail when the list of fields in unique indexes is large because
+            there is a 128 character limit on index names. So, here we create a custom index name which is the md5
+            of the index name decided by create_compound_indexes.
+            """
+            unique_field_names = ''.join(map(lambda (name, direction): str(name) + str(direction), make_index_list(indexes)))
```

Comment:
Please break this one liner into simpler bits.
What's `indexes` like?
What's the output of: `make_index_list`?
Also, please make sure that we sort the list before concatenating and computing hash.
Sorting ensures that we're able to recreate md5 hashes given field names irrespective of order.